src/Components/test/E2ETest/ServerExecutionTests/ComponentHubInvalidEventTest.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.E2ETest;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.AspNetCore.SignalR.Client;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETests.ServerExecutionTests
+{
+    public class ComponentHubInvalidEventTest : IgnitorTest<AspNetSiteServerFixture>
+    {
+        public ComponentHubInvalidEventTest(AspNetSiteServerFixture serverFixture, ITestOutputHelper output)
+            : base(serverFixture, output)
+        {
+        }
+
+        protected override void InitializeFixture(AspNetSiteServerFixture serverFixture)
+        {
+            serverFixture.BuildWebHostMethod = TestServer.Program.BuildWebHost;
+        }
+
+        protected async override Task InitializeAsync()
+        {
+            var rootUri = ServerFixture.RootUri;
+            Assert.True(await Client.ConnectAsync(new Uri(rootUri, "/subdir")), "Couldn't connect to the app");
+            Assert.Single(Batches);
+
+            await Client.SelectAsync("test-selector-select", "BasicTestApp.CounterComponent");
+            Assert.Equal(2, Batches.Count);
+        }
+
+        [Fact]
+        public async Task DispatchingAnInvalidEventArgument_DoesNotProduceWarnings()
```

Comment:
These are not new tests, they were split out from some of the others to commonize their setup code.